include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -929,6 +1112,8 @@ namespace VectorTools
                 const hp::QCollection<dim-1>               &q_boundary,
                 const bool                                  project_to_boundary_first)
   {
+    Assert((dynamic_cast<const parallel::distributed::Triangulation<dim,spacedim>* > (&(dof.get_triangulation()))==0),
```

Comment:
i think this assert is out of place, it not a _distributed_ `project`, right? Whereas you already have an assert in `do_project_distributed`.